src/js_sync.hpp

```diff
@@ -912,6 +985,53 @@ void SyncClass<T>::populate_sync_config(ContextType ctx, ObjectType realm_constr
         }
         config.sync_config->stop_policy = session_stop_policy;
 
+        // Client reset
+        //
+        // i)  manual: the error handler is called with the proper error code and a client reset is initiated
+        // ii) discardLocal: the sync client handles it but notifications are send before and after
+        //
+        // The default setting is manual
+
+        const std::string client_reset_manual = "manual";
+        const std::string client_reset_discard_local = "discardLocal";
+        config.sync_config->client_resync_mode = realm::ClientResyncMode::Manual;
+        ValueType client_reset_value = Object::get_property(ctx, sync_config_object, "clientReset");
+        if (!Value::is_undefined(ctx, client_reset_value)) {
+            auto client_reset_object = Value::validated_to_object(ctx, client_reset_value);
+            ValueType client_reset_mode_value = Object::get_property(ctx, client_reset_object, "mode");
+            if (!Value::is_undefined(ctx, client_reset_mode_value)) {
+                std::string client_reset_mode = Value::validated_to_string(ctx, client_reset_mode_value, "mode");
+                if (client_reset_mode == client_reset_manual) {
+                    config.sync_config->client_resync_mode = realm::ClientResyncMode::Manual;
+                }
+                else if (client_reset_mode == client_reset_discard_local) {
+                    config.sync_config->client_resync_mode = realm::ClientResyncMode::DiscardLocal;
+                }
+                else {
+                    throw std::invalid_argument("Unknown argument for clientReset.mode");
```

Comment:
I think we could print the invalid argument and also suggest the valid options 'manual' and 'discardLocal'